SonarQube.Common/CommandLine/ArgumentInstance.cs

```diff
@@ -27,8 +29,56 @@ public ArgumentInstance(ArgumentDescriptor descriptor, string value)
             this.value = value;
         }
 
+        #region Data
+
         public ArgumentDescriptor Descriptor { get { return this.descriptor; } }
 
         public string Value { get { return this.value; } }
+
+
+        #endregion
+
+        #region Static methods
+
+        public static bool TryGetArgument(string id, IEnumerable<ArgumentInstance> arguments, out ArgumentInstance instance)
+        {
+            if (string.IsNullOrWhiteSpace(id))
+            {
+                throw new ArgumentNullException("id");
+            }
+            if (arguments == null)
+            {
+                throw new ArgumentNullException("arguments");
+            }
+
+            instance = arguments.FirstOrDefault(a => ArgumentDescriptor.IdComparer.Equals(a.Descriptor.Id, id));
+            return instance != null;
+        }
+
+        public static bool TryGetArgumentValue(string id, IEnumerable<ArgumentInstance> arguments, out string value)
+        {
+            if (string.IsNullOrWhiteSpace(id))
```

Comment:
@duncanpMS P3: I'd even delegate the checking of arguments to `TryGetArgument` and not duplicate